source/index.ts

```diff
@@ -385,6 +385,231 @@ const predicateOnArray = (method: ArrayMethod, predicate: Predicate, values: unk
+	iterable = 'Iterable',
+	asyncIterable = 'AsyncIterable',
+	nativePromise = 'native Promise',
+	urlString = 'string with a URL',
+	truthy = 'truthy',
+	falsy = 'falsy',
+	nan = 'NaN',
+	primitive = 'primitive',
+	integer = 'integer',
+	safeInteger = 'integer',
+	plainObject = 'plain object',
+	arrayLike = 'array-like',
+	typedArray = 'TypedArray',
+	domElement = 'Element',
+	nodeStream = 'Node.js Stream',
+	infinite = 'infinite number',
+	emptyArray = 'empty array',
+	nonEmptyArray = 'non-empty array',
+	emptyString = 'empty string',
+	nonEmptyString = 'non-empty string',
+	emptyStringOrWhitespace = 'empty string or whitespace',
+	emptyObject = 'empty object',
+	nonEmptyObject = 'non-empty object',
+	emptySet = 'empty set',
+	nonEmptySet = 'non-empty set',
+	emptyMap = 'empty map',
+	nonEmptyMap = 'non-empty map',
+
+	evenInteger = 'even integer',
+	oddInteger = 'odd integer',
+
+	directInstanceOf = 'T',
+	inRange = 'in range',
+
+	any = 'predicate returns truthy for any value',
+	all = 'predicate returns truthy for all values',
+}
+
+// Type assertions have to be declared with an explicit type.
+interface Assert {
+	// Unknowns.
+	undefined: (value: unknown) => asserts value is undefined;
+	string: (value: unknown) => asserts value is string;
+	number: (value: unknown) => asserts value is number;
+	bigint: (value: unknown) => asserts value is bigint;
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	function_: (value: unknown) => asserts value is Function;
+	null_: (value: unknown) => asserts value is null;
+	class_: (value: unknown) => asserts value is Class;
+	boolean: (value: unknown) => asserts value is boolean;
+	symbol: (value: unknown) => asserts value is symbol;
+	numericString: (value: unknown) => asserts value is string;
+	array: <T = unknown>(value: unknown) => asserts value is T[];
+	buffer: (value: unknown) => asserts value is Buffer;
+	nullOrUndefined: (value: unknown) => asserts value is null | undefined;
+	object: (value: unknown) => asserts value is Record<string, unknown>;
+	iterable: <T = unknown>(value: unknown) => asserts value is Iterable<T>;
+	asyncIterable: <T = unknown>(value: unknown) => asserts value is AsyncIterable<T>;
+	generator: (value: unknown) => asserts value is Generator;
+	nativePromise: <T = unknown>(value: unknown) => asserts value is Promise<T>;
+	promise: <T = unknown>(value: unknown) => asserts value is Promise<T>;
+	generatorFunction: (value: unknown) => asserts value is GeneratorFunction;
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	asyncFunction: (value: unknown) => asserts value is Function;
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	boundFunction: (value: unknown) => asserts value is Function;
+	regExp: (value: unknown) => asserts value is RegExp;
+	date: (value: unknown) => asserts value is Date;
+	error: (value: unknown) => asserts value is Error;
+	map: <TKey = unknown, TValue = unknown>(value: unknown) => asserts value is Map<TKey, TValue>;
```

Comment:
```suggestion
	map: <Key = unknown, Value = unknown>(value: unknown) => asserts value is Map<Key, Value>;
```

Reply:
What do you think about adding generic type parameters to some of the `is` methods too?

Reply:
Would be nice to propagate some more typings for the cases when typescript already knows about them. Separate pull request, right?

Reply:
> Separate pull request, right?

Yes, would be great :)